server/src/users/use-cases/updatePassword/update-password.ts

```diff
@@ -0,0 +1,43 @@
+import { BadRequestException, Inject, Injectable } from '@nestjs/common';
+import { hash } from 'bcrypt';
+import zxcvbn from 'zxcvbn';
+
+import { UpdatePasswordDTO } from '@users/dto/update-password.dto';
+import { User } from '@users/infra/entities/user.entity';
+import { IUsersRepository } from '@users/interfaces/IUsersRepository';
+
+Injectable();
+export class UpdatePassword {
+  constructor(
+    @Inject(IUsersRepository)
+    private readonly usersRepository: IUsersRepository,
+  ) {}
+
+  async execute({
```

Comment:
E se o usuário não passar id, password e nem new_password? Validar antes dos primeiros ifs

Reply:
o DTO já está forçando a passar estes valores

Reply:
Lembra que a possibilidade de não burlar é falha, então coloque as validações igual está feito nos outros use-cases, pois é melhor ter mais uma camada que seja de segurança